nupic/research/frameworks/continual_learning/maml_utils.py

```diff
@@ -0,0 +1,150 @@
+#  Numenta Platform for Intelligent Computing (NuPIC)
+#  Copyright (C) 2020, Numenta, Inc.  Unless you have an agreement
+#  with Numenta, Inc., for a separate license for this software code, the
+#  following terms and conditions apply:
+#
+#  This program is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU Affero Public License version 3 as
+#  published by the Free Software Foundation.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#  See the GNU Affero Public License for more details.
+#
+#  You should have received a copy of the GNU Affero Public License
+#  along with this program.  If not, see http://www.gnu.org/licenses.
+#
+#  http://numenta.org/licenses/
+#
+
+# These are extracted from:
+# https://s3.amazonaws.com/deep-learning-models/image-models/imagenet_class_index.json
+
+from copy import deepcopy
+
+import torch
+
+from nupic.research.frameworks.pytorch.model_utils import get_parent_module
+
+
+def clone_module(module, keep_as_reference=None):
```

Comment:
It looks like Michaelangelo made many modifications to the function in order to fit our implementation. It's not just a simple copy.

Reply:
If that's the case, could you mark the difference? 
Or if possible, contribute the changes back to the original library?
The code below has many `TODO` and references github issues that are not related to our project.
It is easier to maintain a clean library dependency than copied, pasted and modified library code.

Reply:
I'm going to remove `clone_module` since it doesn't get called. The other function, `clone_model`, isn't a copy of the learn2learn code, so we won't need to use that as a dependency nor contribute to their library.